lib/rubocop/cop/rspec/described_class_module_wrapping.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Avoid opening modules and defining specs within them.
+      #
+      # @example
+      #   # bad
+      #   module MyModule
+      #     RSpec.describe MyClass do
+      #       # ...
+      #     end
+      #   end
+      #
+      #   # good
+      #   RSpec.describe MyModule::MyClass do
+      #     # ...
+      #   end
+      #
+      # @see https://github.com/rubocop-hq/rubocop-rspec/issues/735
+      class DescribedClassModuleWrapping < Cop
+        MSG = 'Avoid opening modules and defining specs within them.'
+
+        def_node_search :find_rspec_blocks,
+                        ExampleGroups::ALL.block_pattern
```

Comment:
I tend to think that this might detect an offence in the following code:
```ruby
module M
  describe 'something' do
    # ...
  end
end
```
Notice the example group methods with no explicit receiver.
This is quite minor considering the fact that RSpec cops are limited to a specific set of folders, and I've never seen any of those methods being used in the modules.

However, since it's just Ruby, it's pretty valid to extract some common code with example groups into a module. It's uncommon due to the existence of shared examples, but not something that I would never expect to see in the wild. E.g.:

```
module SomeCommonExamples
  describe 'some common things' do
    it 'does this' do
      # ...
    end
  end
end

RSpec.describe SomeClass do
  include SomeCommonExamples
end
```

Reply:
The cop is disabled by default, so I think it’s not going to be a problem. One could exclude the cop for all files in a folder containing such re-usable examples.